src/app/project/ontology/ontology.component.ts

```diff
@@ -193,36 +192,91 @@ export class OntologyComponent implements OnInit {
+                        // one ontology is selected:
+                        // get all information to display this ontology
+                        // with all classes, properties and connected lists
+                        this.loadOntology = true;
+                        this.ontology = this.ontologies.find(onto => onto.id === this.ontologyIri);
+                        this._cache.set('currentOntology', this.ontology);
+                        this.ontology = response;
+
+                        // grab the onto class information to display
+                        const allOntoClasses = response.getAllClassDefinitions();
+
+                        // reset the ontology classes
+                        this.ontoClasses = [];
+
+                        // display only the classes which are not a subClass of Standoff
+                        allOntoClasses.forEach(resClass => {
+                            const splittedSubClass = resClass.subClassOf[0].split('#');
+                            if (!splittedSubClass[0].includes(Constants.StandoffOntology) && !splittedSubClass[1].includes('Standoff')) {
+                                this.ontoClasses.push(resClass);
+                            }
+                        });
+                        // sort classes by label
+                        // --> TODO: add sort functionallity to the gui
+                        this.ontoClasses = this._sortingService.keySortByAlphabetical(this.ontoClasses, 'label');
+
+                        // grab the onto properties information to display
+                        const allOntoProperties = response.getAllPropertyDefinitions();
+
+                        // reset the ontology properties
+                        this.ontoProperties = [];
+
+                        // display only the properties which are not a subjectType of Standoff
+                        allOntoProperties.forEach(resProp => {
+                            const standoff = (resProp.subjectType ? resProp.subjectType.includes('Standoff') : false);
+                            if (resProp.objectType !== Constants.LinkValue && !standoff) {
+                                this.ontoProperties.push(resProp);
+                            }
+                        });
+                        // sort properties by label
+                        // --> TODO: add sort functionallity to the gui
+                        this.ontoProperties = this._sortingService.keySortByAlphabetical(this.ontoProperties, 'label');
+                    }
+                },
+                (error: ApiResponseError) => {
+                    this._errorHandler.showMessage(error);
+                }
+            );
             await callback(ontologies[i]);
         }
     }
 
+    async loadAndCache(ontologies: OntologyMetadata[]) {
+        await this.asyncForEach(ontologies, async (onto: OntologyMetadata) => {
+            await this.waitFor(200);
```

Comment:
Can you clarify why this method is needed? I'm not a huge fan of `waitFor` and we have stated in our [design doc](https://docs.google.com/document/d/1SIVMxoV0N8xYqhVk_0hhuqSuUF47D1YYrDESqdmipfM/edit#heading=h.jjkobsuiogsh) that we shouldn't use setTimeout in production code. How will this behave with a rather slow internet connection?

Reply:
Whats happens if wait is over but data is still not ready?

Reply:
Yes I know, that we have this rule in our conventions. But I have no idea how to solve this async requests in another way. I found this solution in a tutorial. Maybe someone of you has a better idea?

Reply:
I'll pull the changes from main and check it out

Reply:
One way is:
1. set Boolean variable to false
2. once your data is ready, set this variable to true
3. add watch on the boolean variable defined above and so when it becomes true, call the function to want run after the Ajax data is ready

Reply:
I'll try that...

Reply:
idk if you've already found this @kilchenmann but the issue isn't that the data isn't arriving in time from the API, it's that `this.ontologies` isn't modified in time for the length check within `loadAndCache` so its length is always 0.

Without `waitFor`:
<img width="520" alt="Screen Shot 2021-02-26 at 16 44 37" src="https://user-images.githubusercontent.com/60604010/109322630-cef8df00-7852-11eb-818e-4884daba69a4.png">

With `waitFor`:
<img width="507" alt="Screen Shot 2021-02-26 at 16 47 18" src="https://user-images.githubusercontent.com/60604010/109322657-d6b88380-7852-11eb-8f03-980696c54acb.png">

I think it's due to your callback. Admittedly I'm quite horrible at dealing with async/await things but I'm quite certain it's an issue with your callback getting triggered earlier than it should.

Reply:
Thank you @mdelez 
I'm open for better solution than async/await. At the moment I'm lost. Even with the solution from @waychal I'm not successful. I'm pretty sure I'm doing something wrong, but I have no idea what and how to solve it 😞

Reply:
ok, I came up with another solution without async/await. See my last commit (d174ba2).